src/Shared/PackageManagers/MavenProjectManager.cs

```diff
@@ -315,48 +459,67 @@ private static MavenArtifactType GetMavenArtifactType(string fileName)
+                }
 
-            // Break the version content down into its individual lines, and then find the one that represents the
-            // intended version.
-            string? versionContent = document.QuerySelector("#contents").TextContent
-                .Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries)
-                .SingleOrDefault(versionText => versionText.StartsWith($"{packageVersion}/"));
+                HtmlParser parser = new();
+                AngleSharp.Html.Dom.IHtmlDocument document = await parser.ParseDocumentAsync(html);
 
-            if (versionContent == null)
-            {
-                return null;
-            }
+                // Break the version content down into its individual lines, and then find the one that represents the
+                // intended version.
+                string? versionContent = document.QuerySelector("#contents").TextContent
+                    .Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries)
+                    .SingleOrDefault(versionText => versionText.StartsWith($"{packageVersion}/"));
+
+                if (versionContent == null)
+                {
+                    return null;
+                }
 
-            // Split the version content into its individual parts.
-            // [0] - The version
-            // [1] - The date it was published
-            // [2] - The time it waas published
-            // [3] - The download count
-            string[] versionParts = versionContent.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
-            if (DateTime.TryParse($"{versionParts[1]} {versionParts[2]}", out DateTime publishDateTime))
+                // Split the version content into its individual parts.
+                // [0] - The version
+                // [1] - The date it was published
+                // [2] - The time it waas published
+                // [3] - The download count
+                string[] versionParts = versionContent.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+                if (DateTime.TryParse($"{versionParts[1]} {versionParts[2]}", out DateTime publishDateTime))
+                {
+                    return publishDateTime;
+                }
+            }
+            else if (upstream == MavenSupportedUpstream.GoogleMavenRepository)
             {
-                return publishDateTime;
+                var baseUrl = $"{upstream.GetDownloadRepositoryUrl().EnsureTrailingSlash()}{packageNamespace}/{packageName}/{packageVersion}/{packageName}-{packageVersion}.pom";
+                HttpResponseMessage response = await httpClient.GetAsync(baseUrl);
+
+                // Try to get the "Last-Modified" header
+                if (response.Content.Headers.TryGetValues("Last-Modified", out var values))
+                {
+                    var lastModified = DateTime.Parse(values.First());
+                    return lastModified;
+                }
```

Comment:
For the handful of repositories that I have tested, if this header is provided (some don't provide it), then it seems accurate.

I wonder whether this should be a fallback approach for any repository type, if the directory index approach doesn't work? We could still have the Google Maven repository skip the directory index strategy since we know that won't work

Reply:
Resolving this comment in favor of continuing this conversation in the other comment thread.